### ISSUE TITLE
Cookie handling broken in PHP 7.1

### ISSUE THREAD

**Issue by GawainLynch** [bolt/bolt#5895]
Session cookies not getting sent.

Oh, this could be fun.

@CarsonF I am guessing you don't have an easy 7.1 set up to get going, so let's coordinate on this one in a couple of days+

**Comment by CarsonF** [comment]
Works for me...

**Comment by GawainLynch** [comment]
Did you flush existing cookies? Just set up a fresh site, on a fresh FPM … yeah, after doing said thing mentioned in Slack… and same result.

**Comment by CarsonF** [comment]
Yeah I flushed existing cookies

**Comment by GawainLynch** [comment]
tl;dr The problem is not sessions

OK, so Remi just pushed builds for 7.1-RC4 and out of laziness I initially tried using the built-in web server … which worked … then switched to an FPM and the problem remains.

What I have found so far (will xdebug later, as promised) is that the session files are created, and include the flashes that I would expect … including "You've been logged on successfully." … so this is **not sessions** that are to blame.

What is not happening under the FPM is **cookies** getting set

#### PHP 7.1-RC4
![image](https://cloud.githubusercontent.com/assets/1427081/19467348/05ae8c18-9511-11e6-83e9-741045f4e4c5.png)

#### PHP 7.0.12
![image](https://cloud.githubusercontent.com/assets/1427081/19467332/f547f8be-9510-11e6-9bc2-eb25610d62df.png)

**Comment by GawainLynch** [comment]
Oh the irony … so tried to run though xdebug, and Lazy Strikes Back, I turned off deprecation notices 'cause RandomLib noise … will you guess what happened … I logged in (in and out several times, FWIW), on current HEAD for both `release/3.2` & `release/3.3` sucessfully.

Curious to see what was going on, I started the debugging session, and the moment that xdebug was assembling frame data, the FPM crashed … so I turned deprecation back **on** and the problem returned!

I am going to point the finger at xdebug at last release (2.4.1) was August 1st, so it is reasonable to assume that Derick is overloaded again in real life and hasn't had the time to update for 7.1 … maybe.

I'll retest when the next xdebug release is tagged, and react accordingly as/if needed.
